src/proc.ts

```diff
@@ -163,3 +169,94 @@ export function execute(command: string,
   });
   return {child, result};
 }
+
+/**
+ * Execute a command and return the result
+ * @param command The binary to execute
+ * @param args The arguments to pass to the binary
+ * @param outputConsumer An output consumer for the command execution
+ * @param options Additional execution options
+ *
+ * @note Output from the command is accumulated into a single buffer: Commands
+ * which produce a lot of output should be careful about memory constraints.
+ */
+export function executeFile(command: string,
+                            args: string[],
+                            outputConsumer?: OutputConsumer|null,
+                            options?: ExecFileOptions): Subprocess {
+  if (options && options.silent !== true) {
+    log.info('Executing command: '
+             // We do simple quoting of arguments with spaces.
+             // This is only shown to the user,
+             // and doesn't have to be 100% correct.
+             + [command]
+                   .concat(args)
+                   .map(a => a.replace('"', '\"'))
+                   .map(a => /[ \n\r\f;\t]/.test(a) ? `"${a}"` : a)
+                   .join(' '));
+  }
+  if (!options) {
+    options = {};
+  }
+  const final_env = util.mergeEnvironment(process.env as EnvironmentVariables, options.environment || {});
+
+  const exec_opts: proc.ExecOptionsWithStringEncoding
+      = {env: final_env, shell: options.shell, encoding: options.encoding || 'utf8'};
+
+  if (options && options.cwd) {
+    exec_opts.cwd = options.cwd;
+  }
+
+  const child: proc.ChildProcess = proc.execFile(command, args, exec_opts);
```

Comment:
Use the `callback` signature of [the `child_process.execFile()` API](https://nodejs.org/api/child_process.html#child_process_child_process_execfile_file_args_options_callback). We don't need the output streaming abilities and `OutputConsumer` stuff.